tracer/src/Datadog.Trace/Sampling/RateLimiter.cs

```diff
@@ -31,9 +33,12 @@ internal class RateLimiter : IRateLimiter
         private int _previousWindowChecks = 0;
         private int _previousWindowAllowed = 0;
 
-        public RateLimiter(int? maxTracesPerInterval)
+        public RateLimiter(int? maxTracesPerInterval, bool warnOnDrop = true, bool setAppSecMetric = false)
         {
             _maxTracesPerInterval = maxTracesPerInterval ?? 100;
+            _warnOnDrop = warnOnDrop;
+            _setAppSecMetric = setAppSecMetric;
```

Comment:
It would be better not to tie AppSec and RateLimiter together. To keep it generic you could maybe pass 2 actions,a `OnDropAction` that would set the appsec metric in your case, and one `OnAllSpanAction` that would be set for the current rate limiter use case.

Reply:
I implemented this as an abstract class with two methods to override, rather than passing actions to the constructor.